### PR TITLE
health, DNS stuff

### PR DESCRIPTION
**Pull request by bradfitz**
2 commits

### CODE REVIEW

File: health/health.go [hunk 1]
```diff
@@ -67,16 +68,82 @@ const (
...
+	return warnOptFunc(func(w *Warnable) {
+		w.debugFlag = name
+	})
+}
+
+type warnOptFunc func(*Warnable)
+
+func (f warnOptFunc) mod(w *Warnable) { f(w) }
+
+// Warnable is a health check item that may or
```
Comment (maisem) [inline health/health.go:104]: Incomplete sentence

Comment (DentonGentry) [inline health/health.go:104]: Comment appears to be truncated.

**Review by maisem — APPROVED** [review]

File: net/dns/direct.go [hunk 2]
```diff
@@ -336,7 +337,7 @@ func (m *directManager) checkForFileTrample() {
 		show = show[:1024]
 	}
 	m.logf("trample: resolv.conf changed from what we expected. did some other program interfere? current contents: %q", show)
-	// TODO(bradfitz): register with health package that something is wrong
+	warnTrample.Set(errors.New("Linux DNS config not ideal. /etc/resolv.conf overwritten. See https://tailscale.com/s/dns-fight"))
```
Comment (DentonGentry) [inline net/dns/direct.go:340]: We'll use https://tailscale.com/s/* to redirect to KB articles?

**Review by DentonGentry — APPROVED** [review]

File: net/dns/direct.go (hunk 2 above)
Comment (bradfitz) [reply net/dns/direct.go:340]: Yeah, we use /s/ already in a handful of places in the client, as we can't update old clients in the field but we can update our redirect.txt file :)

**Pull request merged**